assets/app/scripts/controllers/newfromtemplate.js

```diff
@@ -188,17 +196,38 @@ angular.module('openshiftConsole')
           );
         };
 
-        DataService.get("templates", name, {namespace: (namespace || $scope.projectName)}).then(
-          function(template) {
-            $scope.template = template;
-            $scope.templateImages = imageItems(template);
-            $scope.hasParameters = $scope.template.parameters && $scope.template.parameters.length > 0;
-            $scope.templateUrl = template.metadata.selfLink;
-            template.labels = template.labels || {};
-          },
-          function() {
-            Navigate.toErrorPage("Cannot create from template: the specified template could not be retrieved.");
-          });
+        function setTemplateParams() {
+          $scope.templateImages = imageItems($scope.template);
+          $scope.template.labels = $scope.template.labels || {};
+        }
+
+        // Missing namespace indicates that the template should be received from from the 'CachedTemplateService'.
+        // Otherwise get it via GET call. 
+        if (!namespace) {
+          $scope.template = CachedTemplateService.getTemplate();
+          // In case the template can be loaded from 'CachedTemaplteService', show an alert and disable "Create" button.
+          if (_.isEmpty($scope.template)) {
+            $scope.alerts["process"] =
+              {
+                type: "error",
+                message: "Unable to load the Template.",
+                details: "Not found in cache."
+              };
+            $scope.templateForm.$setValidity("",false);
+            return;
```

Comment:
We should navigate to the error page here, or improve the error message. It currently looks like this.

![openshift_web_console](https://cloud.githubusercontent.com/assets/1167259/14885920/18380c38-0d1c-11e6-8cc7-b44feb87a70d.png)